### DISCUSSION TITLE
article/lexical-state-updates

### DISCUSSION THREAD

**Discussion by giscus** [DaniGuardiola/dio.la#9] [Article comments]
# article/lexical-state-updates

Lexical, explained - part 3.

https://drafts.dio.la/article/lexical-state-updates

**Comment by mohmed98** [comment]
Reallu helpfull, thank you :pray:

**Comment by marcusglowe** [comment]
had no idea about the { discrete: true } arg - wow that's gonna simplify a lot of our code.

**Reply by DaniGuardiola** [reply]
note that this should probably be only used in the server side for most cases - if you need it on the client it's likely you're working against lexical in some way, although I've personally used it as an escape hatch sometimes too

**Comment by felipe-muner** [comment]
🚀🚀

**Comment by sakib412** [comment]
Wow! you nailed it. This will help me a lot as I am building a no code web builder reconciler to track change. 
Thanks a ton!!!

**Comment by tshwangq** [comment]
gr8 writing. really enjoying reading.  I think anyone trying to build with Lexical should read this first.

**Comment by asim-headout** [comment]
This should be a shining example of how simplicity can simply communicate even the most complex subjects. Very well-written. Thanks, Dani.

**Comment by intra-galactic** [comment]
This was super helpful. I'll admit that a few bugs could have been avoided if I had found this article sooner 😅
